### PR TITLE
"Add get auth/profile, without test"

### PR DESCRIPTION
**Pull request by luisc68**

### CODE REVIEW

File: src/controllers/tests/auth.profile.test.ts
Comment (wmartzh) [inline src/controllers/tests/auth.profile.test.ts:1]: Just test services, so delete this one

File: src/controllers/auth.controller.ts [hunk 1]
```diff
@@ -49,6 +50,17 @@ class AuthController extends BaseController {
       this.errorHandler(res, error);
     }
   }
+  async profile(req: Request, res: Response) {
+    try {
+      const token = req.headers.authorization?.split(" ")[1];
```
Comment (wmartzh) [inline src/controllers/auth.controller.ts]: This is bad implementation, we just use auth middleware

File: src/services/auth.service.ts [hunk 2]
```diff
@@ -59,6 +59,20 @@ class AuthService {
...
+    const { SECRET_KEY } = process.env;
+    const promise: (token: string, key: string) => Promise<any> =
+      promisify(jwt.verify).bind(jwt);
+    const payload = await promise(Token, SECRET_KEY || "");
+    const user = await userService.findUserByEmail(payload.email);
+
+    if (!user) {
+      throw new HttpError("User doesn't exist", 404);
+    }
+    const { password, ...rest } = user;
```
Comment (wmartzh) [inline src/services/auth.service.ts]: exclude password on querying to the database

File: src/routes/auth.ts [hunk 3]
```diff
@@ -5,4 +5,5 @@ export default Router()
 
   .post("/register", (req, res) => authController.register(req, res))
   .post("/login", (req, res) => authController.login(req, res))
-  .post("/refreshToken", (req, res) => authController.resfreshToken(req, res));
\ No newline at end of file
+  .post("/refreshToken", (req, res) => authController.resfreshToken(req, res))
+  .get("/profile", (req, res) => authController.profile(req, res));
```
Comment (wmartzh) [inline src/routes/auth.ts]: Add auth middleware on route

File: src/controllers/auth.controller.ts (hunk 1 above)
Comment (luisc68) [reply src/controllers/auth.controller.ts]: implementantion corrected in the new commit

File: src/routes/auth.ts (hunk 3 above)
Comment (luisc68) [reply src/routes/auth.ts]: middleware added

File: src/controllers/tests/auth.profile.test.ts
Comment (luisc68) [reply src/controllers/tests/auth.profile.test.ts:1]: ok, deleted.

**Review by wmartzh — APPROVED** [review]

**Pull request merged**